apps/web/src/components/Shared/Login/WalletSelector.tsx

```diff
@@ -105,45 +80,32 @@ const WalletSelector: FC<Props> = ({ setHasConnected, setHasProfile }) => {
     } catch (error) {
       console.error(error);
     } finally {
-      setLoading(false);
       if (!keepModal) {
         setShowAuthModal(false);
       }
     }
   };
 
-  return activeConnector?.id ? (
+  useEffect(() => {
+    if (chain?.id === CHAIN_ID && connector?.id && isConnected) {
+      handleSign();
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [isConnected, chain]);
+
+  return (
     <div className="space-y-3">
       <div className="space-y-2.5">
         {chain?.id === CHAIN_ID ? (
-          <Button
-            disabled={loading}
-            icon={
-              loading ? (
-                <Spinner className="mr-0.5" size="xs" />
-              ) : (
-                <img className="mr-0.5 w-4 h-4" height={16} width={16} src="/lens.png" alt="Lens Logo" />
-              )
-            }
-            onClick={handleSign}
-          >
-            <Trans>Sign-In with Lens</Trans>
-          </Button>
+          <div className="text-sm font-bold text-center flex">
+            <div className="flex">
+              <Spinner size="sm" className="mx-auto mr-2" />
+              <Trans>Signing …</Trans>
+            </div>
+          </div>
         ) : (
           <SwitchNetwork />
```

Comment:
SwitchNetwork may be removed too, if we are fine with letting the wallet take care of the "switch network" dialog (although note that some wallets, like Coinbase Wallet, switches network directly without prompting the user )